### ISSUE TITLE
Improve/expand Record in generated Open API file

### ISSUE THREAD

**Issue by Couto** [asteasolutions/zod-to-openapi#211]
I'm not sure if this is working as intended, but I will explain my problem and my expectations:

We're building a quizz where users have to associate countries with continents, the answers might contain wrong associations ofc.

```ts
const Continents = z
  .enum(['EUROPE', 'AFRICA', 'NORTH_AMERICA'])
  .describe('List of Continents');

const Countries = z
  .enum(['USA', 'CAN', 'GBR'])
  .describe('Country ISO code (Alpha-3)');

export const Geography = z
  .record(Continents, z.object({
    countries: Countries.array(),
  }))
  .openapi('Geography', {
    example: {
      EUROPE: { countries: ['GBR'] },
      NORTH_AMERICA: { countries: ['USA', 'CAN'] },
    },
  });
```

As far as Zod is concerned, this works fine. The problem is the generated Open API file, which could be a bit better (imho). As is, right now, the generated file contains the following:

```yaml
  schemas:
    Geography:
      type: object
      additionalProperties:
        type: object
        properties:
          countries:
            type: array
            items:
              type: string
              enum:
                - USA
                - CAN
                - GBR
              description: Country ISO code (Alpha-3)
            description: Country ISO code (Alpha-3)
        required:
          - countries
      example:
        EUROPE:
          countries:
            - GBR
        NORTH_AMERICA:
          countries:
            - USA
            - CAN

```

However, my expectation of the final Open API file was something like:

```yaml
    Geography:
      type: object
      properties:
        EUROPE:
          type: object
          properties:
            countries:
              type: array
              items:
                type: string
                enum: &a1
                  - USA
                  - CAN
                  - GBR
                description: Country ISO code (Alpha-3)
              description: Country ISO code (Alpha-3)
          required:
            - countries
        AFRICA:
          type: object
          properties:
            countries:
              type: array
              items:
                type: string
                enum: *a1
                description: Country ISO code (Alpha-3)
              description: Country ISO code (Alpha-3)
          required:
            - countries
        NORTH_AMERICA:
          type: object
          properties:
            countries:
              type: array
              items:
                type: string
                enum: *a1
                description: Country ISO code (Alpha-3)
              description: Country ISO code (Alpha-3)
          required:
            - countries
      example:
        EUROPE:
          countries:
            - GBR
        NORTH_AMERICA:
          countries:
            - USA
            - CAN

```

Is my expectation wrong here? I know it's possible to use the `z.object` to achieve the final example, and it works fine with a small number of keys, but codewise the `z.enum` should work? Is this something that can be improved?

**Comment by AGalabov** [comment]
@Couto thank you for bringing this up.

To be completely honest - we've never into the option of providing keys to the record. I would have to check how the schema looks like within zod and come back to you with an answer of whether or not it is possible

**Comment by AGalabov** [comment]
Hey @Couto - I've just opened a PR with pretty much your example in the tests - feel free to take a look. Hopefully we should be able to review, merge and publish it soon.

**Comment by AGalabov** [comment]
@Couto - released as part of [v6.4.0](https://github.com/asteasolutions/zod-to-openapi/releases/tag/v6.4.0)

**Comment by Couto** [comment]
Thank you so much for this, and the quick response.
